impit-node/src/response.rs

```diff
@@ -1,23 +1,43 @@
-use std::collections::HashMap;
+#![allow(clippy::await_holding_refcell_ref, deprecated)]
+use std::{cell::RefCell, collections::HashMap, ops::Deref};
 
 use impit::utils::{decode, ContentType};
-use napi::{bindgen_prelude::Buffer, Env, JsFunction, JsObject, JsString, JsUnknown};
+use napi::{
+  bindgen_prelude::{Buffer, BufferSlice, ReadableStream, Result},
+  Env, JsFunction, JsObject, JsString, JsUnknown,
+};
 use napi_derive::napi;
 use reqwest::Response;
+use tokio_stream::StreamExt;
 
 #[napi]
 pub struct ImpitResponse {
-  bytes: Vec<u8>,
+  bytes: RefCell<Option<Vec<u8>>>,
+  inner: RefCell<Option<Response>>,
   pub status: u16,
   pub status_text: String,
   pub headers: HashMap<String, String>,
   pub ok: bool,
 }
 
+/// Ensures that the response has been read and the bytes are available.
+/// This is not part of the impl ImpitResponse block because it is an async function,
+/// which causes issues with the napi_derive macro.
+///
+/// Note that calling this method will consume the response.
+async fn read_response_bytes(impit: &ImpitResponse) {
+  let mut response = impit.inner.borrow_mut();
+  if let Some(inner_response) = response.take() {
+    let bytes = inner_response.bytes().await.unwrap().to_vec();
+    impit.bytes.replace_with(|_| Some(bytes));
+  } else {
+    panic!("fatal: Response already consumed");
```

Comment:
Can node.js/python bindings handle panics in rust code? I.e., convert them to exceptions?

Reply:
Nope, I think it cannot handle those (you'd have to return `Result<_, napi::Error>` in order to do that). 

Good point - a user calling `.body` twice probably doesn't deserve to have their script killed on spot. I'll revise those calls and likely add some tests for this.

Reply:
I revised the calls, hopefully, the codebase is now more beginner-user-friendly. There are still `panic` macro calls, but only in the most unexpected (and irreparable) states. Resolving for now.